### PR TITLE
more tolerance in reading yaml

### PR DESCRIPTION
**Pull request by mbuerki**
The underlying problem is the following:
Minkindr is machine-precision strict when it comes to the construction of rotation matrices.
Reading/writing rotation matrices to/from file (e.g., to a YAML file) is usually less precise, because much fewer number of digits are used compared to floating point machine precision. 
This occasionally led to check fails when reading in YAML files with in principle correct rotation matrices.

Note that `fromApproximateRotationMatrix()` still checks that the given matrix is a valid rotation matrix, but it's not as numerically strict as the default constructor.

### CODE REVIEW

File: aslam_cv_cameras/src/ncamera-yaml-serialization.cc [hunk 1]
```diff
@@ -73,9 +73,10 @@ bool convert<std::shared_ptr<aslam::NCamera> >::decode(const Node& node,
         return true;
       }
       // This call will fail hard if the matrix is not a rotation matrix.
```
Comment (dymczykm) [inline aslam_cv_cameras/src/ncamera-yaml-serialization.cc]: is this still true?

File: aslam_cv_cameras/src/ncamera-yaml-serialization.cc [hunk 2]
```diff
@@ -73,9 +73,10 @@ bool convert<std::shared_ptr<aslam::NCamera> >::decode(const Node& node,
...
-      aslam::Quaternion q_B_C = aslam::Quaternion(
+      const aslam::Quaternion q_B_C =
+          aslam::Quaternion::fromApproximateRotationMatrix(
```
Comment (dymczykm) [inline aslam_cv_cameras/src/ncamera-yaml-serialization.cc]: what's the precision here? can we still impose a threshold?

File: aslam_cv_cameras/src/ncamera-yaml-serialization.cc (hunk 1 above)
Comment (mbuerki) [reply aslam_cv_cameras/src/ncamera-yaml-serialization.cc]: yes.

File: aslam_cv_cameras/src/ncamera-yaml-serialization.cc (hunk 2 above)
Comment (mbuerki) [reply aslam_cv_cameras/src/ncamera-yaml-serialization.cc]: yes, there is still a threshold. it's 1e-4 instead of 1e-8.

**Pull request merged**